### PR TITLE
Block work

### PR DESCRIPTION
**Pull request by supersaiyansubtlety**
Completes `block` except for a couple vault classes

### CODE REVIEW

**Comment by supersaiyansubtlety** [comment]
~~Please let the conflict hell be over :sob:~~
I got so much `git rebase` practice :smiley:

**Comment by eristhea** [comment]
> ~Please let the conflict hell be over 😭 ~

we torture you so..

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 1]
```diff
@@ -404,7 +405,7 @@ CLASS net/minecraft/unmapped/C_triydqro net/minecraft/block/AbstractBlock
 			ARG 1 state
 			ARG 2 direction
 		METHOD m_pobwdefx exceedsCube ()Z
-		METHOD m_pppprjpr isTranslucent (Lnet/minecraft/unmapped/C_peaveboq;Lnet/minecraft/unmapped/C_hynzadkk;)Z
+		METHOD m_pppprjpr isTransparent (Lnet/minecraft/unmapped/C_peaveboq;Lnet/minecraft/unmapped/C_hynzadkk;)Z
```
Comment (eristhea) [inline mappings/net/minecraft/block/AbstractBlock.mapping:408]: why this change? the mojmap name is `propagatesSkylightDown`, which doesn't necessarily mean full transparency

File: mappings/net/minecraft/block/RedstoneWireBlock.mapping [hunk 2]
```diff
@@ -45,6 +45,11 @@ CLASS net/minecraft/unmapped/C_hbdlqish net/minecraft/block/RedstoneWireBlock
 		ARG 2 random
 		ARG 3 pos
 		ARG 4 color
+		ARG 5 support
+		ARG 6 connection
+		ARG 7 offset
+		ARG 8 randomFactor
+			COMMENT higher values make particles both more likely and further randomly offset
```
Comment (eristhea) [inline mappings/net/minecraft/block/RedstoneWireBlock.mapping:52]:
```suggestion
			COMMENT higher values make particles both more likely and randomly offset further
```

the current wording implies that the offset is more random, i'm assuming you meant it this way

File: mappings/net/minecraft/block/entity/JukeboxBlockEntity.mapping [hunk 3]
```diff
@@ -1,5 +1,25 @@
 CLASS net/minecraft/unmapped/C_qgqgzpda net/minecraft/block/entity/JukeboxBlockEntity
```
Comment (eristhea) [inline mappings/net/minecraft/block/entity/JukeboxBlockEntity.mapping:1]: why `record` instead of `disc` in this class?

File: mappings/net/minecraft/block/entity/SkullBlockEntity.mapping [hunk 4]
```diff
@@ -1,28 +1,51 @@
...
 	METHOD m_stbhfbap clientTick (Lnet/minecraft/unmapped/C_cdctfzbn;Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_txtbiemp;Lnet/minecraft/unmapped/C_vrhaavsg;)V
 		ARG 3 skull
+	METHOD m_uqvkmqng getProfile (Ljava/lang/String;)Ljava/util/concurrent/CompletableFuture;
+		ARG 0 username
+	METHOD m_yjswcdkm (Lnet/minecraft/unmapped/C_iqgmrwok;)V
+		ARG 1 owner
+	METHOD m_yqomcdhv fetchProfile (Ljava/lang/String;Lnet/minecraft/unmapped/C_vmxvaoqe;)Ljava/util/concurrent/CompletableFuture;
+		ARG 1 services
 	METHOD m_yrfbflxq loadOwnerProperties ()V
+	CLASS C_htjtxsnn
```
Comment (eristhea) [inline mappings/net/minecraft/block/entity/SkullBlockEntity.mapping]: very curious why this got written

**Review by eristhea — COMMENTED** [review]
reviewed from web, i'll do a more in-depth one later
i really appreciate all the refactors to unify under `bootstrap` and direct codecs!

File: mappings/net/minecraft/block/AbstractBlock.mapping (hunk 1 above)
Comment (supersaiyansubtlety) [reply mappings/net/minecraft/block/AbstractBlock.mapping:408]: It's for this
```java
protected int getOpacity(BlockState state, BlockView world, BlockPos pos) {
    if (state.isOpaqueFullCube(world, pos)) {
        return world.getMaxLightLevel();
    } else {
        return state.isTransparent(world, pos) ? 0 : 1;
    }
}
```

File: mappings/net/minecraft/block/RedstoneWireBlock.mapping (hunk 2 above)
Comment (supersaiyansubtlety) [reply mappings/net/minecraft/block/RedstoneWireBlock.mapping:52]:
```java
float h = randomFactor - offset;
// ...
float j = offset + h * random.nextFloat();
```
It's "more random" because it's random over a greater range, but it's also more likely to by further.
I'm not sure which wording is clearer.

File: mappings/net/minecraft/block/entity/JukeboxBlockEntity.mapping (hunk 3 above)
Comment (supersaiyansubtlety) [reply mappings/net/minecraft/block/entity/JukeboxBlockEntity.mapping:1]:
```java
public static final String RECORD_ITEM_KEY = "RecordItem";
```

File: mappings/net/minecraft/block/entity/SkullBlockEntity.mapping (hunk 4 above)
Comment (supersaiyansubtlety) [reply mappings/net/minecraft/block/entity/SkullBlockEntity.mapping]: Figured it out:
- I tried to name a `UUID uUID` param `uuid`
- this results in
```
CLASS C_htjtxsnn
	METHOD m_palpyqag (Ljava/util/UUID;)Ljava/util/concurrent/CompletableFuture;
		ARG 1 uuid
```
- then `dropInvalidMappings` results in `CLASS C_htjtxsnn`

**Review by eristhea — APPROVED** [review]

**Review by OroArmor — APPROVED** [review]
Oops thought i approved this earlier
